### PR TITLE
Create docker compose generator

### PR DESCRIPTION
**Pull request by MayoG**
- Created the new compose generator script

- Deleted files that will be generated from the compose generator

- Created a file with functions that is being used by a lot of classes around the project (The reason most of the files changed in this pull request)

### CODE REVIEW

File: compose_generator.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+config_file = get_config_file()
+
+monitoring_system = config_file["monitoring_system"]
+if monitoring_system.lower() == "prometheus":
+    prometheus_handler(pods=config_file["pods"])
+
+# Create the first pod
+pipeline_first_pod = {
+    "container_name": "",
+    "image": "pipert_pipert",
```
Comment (ItayHoresh) [inline compose_generator.py:215]: Why are you giving the image? We build this image every time?

File: compose_generator.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    "build": {
+        "context": ".",
+        "args": {
```
Comment (ItayHoresh) [inline compose_generator.py:218]: Can I pass this args to this scripts?

File: compose_generator.py [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+            "aliases": [
+                "pipert"
+            ]
+        }
+    },
+    "depends_on": [
+        "redis",
+        "base-pipert"
+    ],
+    "environment": {
```
Comment (ItayHoresh) [inline compose_generator.py:234]: Can I pass the environment variables to this script?

File: compose_generator.py (hunk 1 above)
Comment (MayoG) [reply compose_generator.py:215]: We build the image only the first time on the first pod.

File: compose_generator.py (hunk 2 above)
Comment (MayoG) [reply compose_generator.py:218]: No, we need to think where do we add it in the configuration file.

File: compose_generator.py (hunk 3 above)
Comment (MayoG) [reply compose_generator.py:234]: No, same as the args, we need to think where do we add it in the configuration file.

**Pull request merged**